runtime/src/bank/partitioned_epoch_rewards/mod.rs

```diff
@@ -150,7 +150,33 @@ pub(super) struct CalculateRewardsAndDistributeVoteRewardsResult {
 
 pub(crate) type StakeRewards = Vec<StakeReward>;
 
+pub struct KeyedRewardsAndNumPartitions {
+    pub keyed_rewards: Vec<(Pubkey, RewardInfo)>,
+    pub num_partitions: Option<u64>,
+}
+
+impl KeyedRewardsAndNumPartitions {
+    pub fn is_empty(&self) -> bool {
+        self.keyed_rewards.is_empty() && self.num_partitions.is_none()
+    }
+}
+
 impl Bank {
+    pub fn get_rewards_and_num_partitions(&self) -> KeyedRewardsAndNumPartitions {
+        let rewards = self.rewards.read().unwrap();
+        let epoch_rewards_sysvar = self.get_epoch_rewards_sysvar();
+        // If partitioned epoch rewards are active and this Bank is the
+        // epoch-boundary block, populate num_partitions
+        let num_partitions = (epoch_rewards_sysvar.active
+            && (self.block_height().saturating_add(1)
+                == epoch_rewards_sysvar.distribution_starting_block_height))
```

Comment:
How about instead we check if the bank's parent is in the previous epoch? That way this is a bit more future proof and is more directly set to the first block in an epoch

Reply:
Sure, that works